scripts/pdf/generate_pdf.py

```diff
@@ -1,225 +1,94 @@
-    listOfFiles.sort(key=putIndexFirst2)
-    allFiles = list()
-    chapter = 0
-
-    # Iterate over all the entries
-    for entry in listOfFiles:
-        # Create full path
-        fullPath = os.path.join(dirName, entry)
-        # If entry is a directory then get the list of files in this directory
-        if os.path.isdir(fullPath):
-            allFiles = allFiles + getListOfFiles(
-                fullPath, parentChapter + str(chapter) + "."
-            )
-        elif ".mdx" in entry or ".md" in entry:
-            allFiles.append(ToCItem(fullPath, parentChapter + str(chapter)))
-
-        chapter += 1
-    return allFiles
-
-
-def main():
-    dirName = ""
-    try:
-        dirName = sys.argv[1]
-        dirName = re.sub(r"\/$", "", dirName)
-    except BaseException:
-        print("directory not passed in")
-        print("if running from yarn use `yarn build-pdf directory/path/here`")
-        sys.exit(1)
-
-    openPdf = False
-    html = False
-    try:
-        html = sys.argv[2] == "--html"
-        openPdf = sys.argv[2] == "--open"
-    except BaseException:
-        pass
-
-    splitDirName = dirName.split("/")
-    product = splitDirName[2]
-    version = splitDirName[3]
-
-    fullProductPdf = True
-    guide = None
-    if len(splitDirName) > 4:
-        fullProductPdf = False
-        guide = splitDirName[4]
-
-    _file_prefix = f"{dirName}/{product}_v{version}"
-    _doc_prefix = f"{_file_prefix}_documentation"
-    mdxFilePath = f"{_doc_prefix}.mdx"
-    htmlFilePath = f"{_doc_prefix}.html"
-    coverFilePath = f"{_doc_prefix}_cover.html"
-    pdfFilePath = f"{_file_prefix}" "_{}documentation.pdf".format(
-        guide + "_" if guide else ""
-    )
-
-    print(f"{ANSI_BLUE}building {pdfFilePath}{ANSI_STOP}")
-
-    if not os.path.exists(dirName):
-        raise Exception("directory does not exist")
 
-    if os.path.exists(mdxFilePath):
-        os.remove(mdxFilePath)
+def main(args):
+    doc_path, product, version, mdx_file, html_file, cover_file, pdf_file = setup(args)
 
-    # Get the list of all files in directory tree at given path
-    listOfFiles = getListOfFiles(dirName, "")
-    if len(listOfFiles) == 0:
-        raise Exception(f"no files in {dirName}")
-    if fullProductPdf:
-        listOfFiles.pop(0)  # remove base product index page, which are empty
+    files = list_files(doc_path)
+    if len(files) == 0:
+        print(f"{ANSI_YELLOW}skipping {pdf_file}{ANSI_STOP}")
+        return
 
-    for elem in listOfFiles:
-        g = open(elem.filename, "r")
-        for line in g.readlines():
-            if "title: " in line:
-                elem.title = line[7:].replace('"', "").replace("\n", "")
-            pattern = re.compile('div id="(.*?)" class="registered_link"')
+    resource_search_paths = {doc_path}
 
-            tag = pattern.search(line)
-            if tag and len(elem.anchor) == 0:
-                elem.anchor = tag.group(1)
```

Comment:
@josh-heyer @robert-stringer Do either of you know what `tag` was about? Both `title` and `tag` are never used in this script at all.

I'm removing the entire `for` loop since the resulting data isn't used at all.

Reply:
Not sure, but looks very similar to some of the link rewriting logic in the juicer scripts. Hasn't been used since at least the import from Rocket, but if I had to guess it's a relic of an alternative method for generating the ToC page.

Reply:
OK, I'm going to remove this dead code, then.